### DISCUSSION TITLE
How do I modify the generated SQL from the Db2 for i extension

### DISCUSSION THREAD

**Discussion by JonFP** [codefori/vscode-ibmi#739] [Q&A] [answered]
The default SELECT * ... is great for small test files - but sometimes I want to add a WHERE close or something. However, when I do I just get syntax errors even when I know the syntax is fine.

What am I missing?

**Comment by worksofliam** [comment]
I'm curious as to what error you're getting. Are you able to share any screenshots?

**Reply by JonFP** [reply]
![image](https://user-images.githubusercontent.com/13474625/178124493-0ae28c74-fce2-4470-b8b5-26c91bbdab8b.png)
There you go.

If I complete the clause anyway and press the "Play" button I get this:
![image](https://user-images.githubusercontent.com/13474625/178124532-aae7a826-0891-4c28-bbde-e34c64919a22.png)

In fact I get that error even if the syntax is accepted e.g. replace the text with (say) SELECT * FROM PARTNER400.CATEGOR

**Reply by worksofliam** [reply]
Looks like a bug on my end. Will let you know when it's fixed.

**Reply by worksofliam** [reply]
@JonFP 

The validator isn't perfect. It's a Db2 validator and not specific to Db2 for i. I wonder if it was erroring because you had some special characters in your column names (which I don't think Db2 LUW likes?). You can turn off the validator in the VS Code settings for the Db2 extension.

Seems the play button is bugged and I am working on that now, but in the meantime the shortcut (Control / Command + R) does work to run a statement!

**Comment by worksofliam** [comment, marked as answer]
@JonFP Update the Db2 for IBM i extension (vscode-db2i) to 0.2.1 and the problem should be resolved. Thanks!

**Reply by JonFP** [reply]
Seems to still be a bit flakey but it sorta works.  Will try to be more definitive when I've had more time to play.
When I first tried to add a WHEN clause it just produced no output.  Then tried an ORDER BY and that worked. So tried a WHERE again - no joy until in desperation I used double quotes instead of single around the comparison value. That was errored as expected but when I then corrected it back to single quotes the statement ran correctly and continued to do so.  Go figure.

**Comment by JonFP** [comment]
Nope - the names are as vanilla as you can get.


Jon Paris

***@***.***
